app/scripts/directives/overview/mobileClientRow.js

```diff
@@ -27,17 +27,31 @@
     var isServiceInstanceReady = $filter('isServiceInstanceReady');
     var isMobileService = $filter('isMobileService');
     row.installType = '';
+    var serviceInstanceClasses = {};
 
     _.extend(row, ListRowUtils.ui);
 
+    function getServiceInstanceServiceClass(serviceInstance, cb){
```

Comment:
how would we invalidate the cache? seems we should be watching them and if a change happens invalidate?